cli/azd/internal/appdetect/spring_boot.go

```diff
@@ -244,8 +292,47 @@ func detectEventHubsAccordingToSpringCloudStreamKafkaMavenDependency(
 	}
 }
 
+func detectEventHubsAccordingToSpringKafkaMavenDependency(azdProject *Project, springBootProject *SpringBootProject) {
+	var targetGroupId = "org.springframework.kafka"
+	var targetArtifactId = "spring-kafka"
+	if hasDependency(springBootProject, targetGroupId, targetArtifactId) {
+		newDep := AzureDepEventHubs{
+			// eventhubs name is empty here because no configured property
+			EventHubsNamePropertyMap: map[string]string{},
+			UseKafka:                 true,
+			SpringBootVersion:        detectSpringBootVersion(springBootProject.pom),
+			FromDependency:           []DependencyType{SpringKafka},
+		}
+		addAzureDepEventHubsIntoProject(azdProject, newDep)
+		logServiceAddedAccordingToMavenDependency(newDep.ResourceDisplay(), targetGroupId, targetArtifactId)
+	}
+}
+
+func addAzureDepEventHubsIntoProject(
+	azdProject *Project,
+	newDep AzureDepEventHubs) {
+	for index, azureDep := range azdProject.AzureDeps {
+		if azureDep, ok := azureDep.(AzureDepEventHubs); ok {
+			// already have existing dependency
+			for property, eventHubsName := range newDep.EventHubsNamePropertyMap {
+				azureDep.EventHubsNamePropertyMap[property] = eventHubsName
+			}
+			azureDep.FromDependency = append(azureDep.FromDependency, newDep.FromDependency...)
+			azureDep.UseKafka = newDep.UseKafka
```

Comment:
Is it possible that current func executed twice and `newDep.UseKafka=true` in the first time and `newDep.UseKafka=false` in the second time?

Reply:
Good point. After deprecating the `UseKafka` property, it won't be impacted.

Reply:
What's the expected behavior for above case? Will `useKafka` func return `true` or `false`? Is it possible that customer's application use Service Bus for Kafka and Service Bus at the same time in one app?

Reply:
The expected behavior is that this property should be `true` as long as kafka used.
So if it is executed twice and `newDep.UseKafka=true` in the first time and `newDep.UseKafka=false` in the second time, it will return wrong result.
However, after we deprecate this property and use a func to replace it (as mentioned in the comment https://github.com/azure-javaee/azure-dev/pull/85#discussion_r1899961042), we can avoid this problem, just check this by checking the `DependencyTypes`.